apps/aecore/src/aec_sync.erl

```diff
@@ -69,105 +66,62 @@ connect_peer(Uri) ->
-                    end
-            end,
-            fetch_mempool(RemoteUri),
-            ok;
-        _ ->
-            {error, different_genesis_blocks}
-    end.
-
-
-start_sync(Uri, RemoteHash, RemoteDifficulty) ->
-    gen_server:cast(?MODULE, {start_sync, Uri, RemoteHash, RemoteDifficulty}).
+start_sync(PeerId, RemoteHash, RemoteDifficulty) ->
+    gen_server:cast(?MODULE, {start_sync, PeerId, RemoteHash, RemoteDifficulty}).
 
-server_get_missing_blocks(Uri) ->
-    gen_server:cast(?MODULE, {server_get_missing, Uri}).
+server_get_missing_blocks(PeerId) ->
+    gen_server:cast(?MODULE, {server_get_missing, PeerId}).
 
-fetch_mempool(Uri) ->
-    gen_server:cast(?MODULE, {fetch_mempool, Uri}).
+fetch_mempool(PeerId) ->
+    gen_server:cast(?MODULE, {fetch_mempool, PeerId}).
 
-schedule_ping(Uri) ->
-    gen_server:cast(?MODULE, {schedule_ping, Uri}).
+schedule_ping(PeerId) ->
+    gen_server:cast(?MODULE, {schedule_ping, PeerId}).
 
-fetch_next(Uri, HeightIn, HashIn, Result) ->
-    gen_server:call(?MODULE, {fetch_next, Uri, HeightIn, HashIn, Result}).
+fetch_next(PeerId, HeightIn, HashIn, Result) ->
+    gen_server:call(?MODULE, {fetch_next, PeerId, HeightIn, HashIn, Result}).
 
-delete_from_pool(Uri) ->
-    gen_server:cast(?MODULE, {delete_from_pool, Uri}).
+delete_from_pool(PeerId) ->
+    gen_server:cast(?MODULE, {delete_from_pool, PeerId}).
 
 %% Check whether this header is better than previous best
-new_header(Uri, Header, AgreedHeight, Hash) ->
-    gen_server:call(?MODULE, {new_header, Uri, Header, AgreedHeight, Hash}).
+new_header(PeerId, Header, AgreedHeight, Hash) ->
+    gen_server:call(?MODULE, {new_header, PeerId, Header, AgreedHeight, Hash}).
 
 update_hash_pool(Hashes) ->
   gen_server:call(?MODULE, {update_hash_pool, Hashes}).
 
-sync_in_progress(Uri) ->
-   gen_server:call(?MODULE, {sync_in_progress, Uri}).
+sync_in_progress(PeerId) ->
+   gen_server:call(?MODULE, {sync_in_progress, PeerId}).
 
 %%%=============================================================================
 %%% gen_server functions
 %%%=============================================================================
 
-%% When we Ping a node with at least as much difficulty as we have, 
+%% When we Ping a node with at least as much difficulty as we have,
 %% then we are going to sync with it.
 %% We already agree upon the genesis block and need to find the highest common
 %% block we agree upon. We use a binary search to find out.
 %% From the height we agree upon, we start asking for blocks to add to the chain.
-%% 
+%%
 %% When an additional Ping arrives for which we agree upon genesis, we have
 %% the following possibilities:
 %% 1. It has worst top hash than our node, do not include in sync
 %% 2. It has better top hash than our node
-%%    We binary search for a node that we agree upon (could be genesis) 
+%%    We binary search for a node that we agree upon (could be genesis)
```

Comment:
Is "node" the right word here? Is it not the "block" we search for?

Reply:
I just fixed the whitespace issue that Thomas introduced, but yes it looks wrong :-)